src/liboslexec/llvm_util.cpp

```diff
@@ -3950,23 +4038,25 @@ LLVM_Util::op_1st_active_lane_of(llvm::Value* mask)
         // and all types are happy
         intMaskType = type_int8();
         break;
-#if 0  // WIP
-        case 4:
-        {
-            // We can just reinterpret cast a 8 bit mask to a 8 bit integer
-            // and all types are happy
-            intMaskType = type_int8();
+    case 4: {
+        intMaskType = type_int8();
 
-//            extended_int_vector_type = (llvm::Type *) llvm::VectorType::get(llvm::Type::getInt32Ty (*m_llvm_context), m_vector_width);
-//            llvm::Value * wide_int_mask = builder().CreateSExt(mask, extended_int_vector_type);
-//
-//            int_reinterpret_cast_vector_type = (llvm::Type *) llvm::Type::getInt128Ty (*m_llvm_context);
-//            zeroConstant = constant128(0);
-//
-//            llvm::Value * mask_as_int =  builder().CreateBitCast (wide_int_mask, int_reinterpret_cast_vector_type);
-            break;
-        }
-#endif
+        llvm::Value* mask_as_int = mask4_as_int8(mask);
+
+        // Count trailing zeros, least significant
+        llvm::Type* types[] = { intMaskType };
```

Comment:
This looks to repeat the code outside the switch statement, I think you can just reduce this down to
``` 
llvm::Value* mask = mask4_as_int8(mask);
break;
```
and let the existing code after the switch handle the rest vs. repeating it.

Reply:
Fixed